course6/toyapp-adam/scene.cc

```diff
@@ -1,29 +1,30 @@
 #include "scene.h"
 
+#include <displaylib/mesh_util.h>
 #include <displaylib/surface_3d.h>
 #include <util/keyboard.h>
 #include <util/mouse.h>
-#include <cmath>
 
 Scene::Scene(Keyboard& keyboard, Mouse& mouse, Surface3D& surface)
 :	mKeyboard{keyboard},
 	mMouse{mouse},
-	mSurface{surface},
-	mTestCube1{{0.0f, 0.0f, 2.5f}, 0.5f,Display::YELLOW, Display::BLACK},
-	mTestCube2{{1.25f, 0.25f, 2.5f}, 0.5f, Display::GREEN, Display::BLACK},
-	mTestCube3{{-1.25f, -0.25f, 2.5f}, 0.3f, Display::RED, Display::BLACK},
-	mTestCube4{{0.0f, 0.75f, 2.5f}, 0.25f, Display::YELLOW, Display::BLACK},
-	mTestCube5{{0.0f, 1.125f, 2.5f}, 0.125f, Display::YELLOW, Display::BLACK}
+	mSurface{surface}
 {}
 
-void Scene::Update(double totalElapsedSeconds)
+void Scene::LoadMesh(const std::string& filePath)
 {
-	using Math::Vector3D;
+	mMesh = Display::CreateMeshFromObjFile(filePath);
+	mMesh.NormalizeScale();
```

Comment:
I don't like this "normalise scale" thing. It mutates the mesh to fit your viewport. It should be the other way around.
Maybe the mesh should have a method that returns AABB, and then you can adjust your camera so that all of the mesh is visible?